improver_tests/acceptance/acceptance.py

```diff
@@ -78,13 +82,13 @@ def cli_name_with_dashes(dunder_file):
 
 def kgo_recreate():
     """True if KGO should be re-created"""
-    return "RECREATE_KGO" in os.environ
+    return RECREATE_DIR in os.environ
```

Comment:
The prompt for user input was previously implemented in the `bin/improver-tests` wrapper script rather than inside the tests themselves. I can't think of a good or sensible way to do this warning prompt from within test functions themselves, which are disconnected from being able to interact with the terminal/keyboard.

I've changed the `recreate_if_needed` function to exit if attempting to overwrite files in-place (as opposed to putting the recreated files in a separate directory). This puts a bit of a safety barrier around accidentally overwriting KGOs.